expo/ticketing/models.py

```diff
@@ -146,7 +157,167 @@ class Transaction(models.Model):
     reference = models.CharField(max_length=30)
     payment_source = models.CharField(max_length=30)
     import_date = models.DateTimeField(auto_now=True)
-    
+
     def __unicode__(self):
         return '%s (%s)' % (self.reference, self.purchaser)
-    
+
+
+class CheckListItem(models.Model):
+    '''
+    This is a physical item that we can give away at the registration desk
+    Examples:  a badge, a wristband, a goodie bag, a guidebook, a release
+    form, etc.  It may or may not be labeled for a specific users (for example,
+    a badge has a name on it)
+    '''
+    description = models.CharField(max_length=50, unique=True)
+
+    def __unicode__(self):
+        return unicode(self.description)
+
+
+class EligibilityCondition(models.Model):
+    '''
+    This is the paremt class connecting the conditions under which a
+    CheckListItem can be given to the conditions themselves.
+
+    Conditions are logically additive unless eliminated by exclusions.
+    So if 3 conditions give an item with no exclusion, then the individual
+    gets 3 items.
+
+    TO Discuss (post expo) - consider making this abstract and using content
+    types to link the exclusion foreign key to abstract cases of this class.
+    '''
+    checklistitem = models.ForeignKey(
+        CheckListItem,
+        related_name="%(app_label)s_%(class)s")
+
+    def no_exclusion(self, held_tickets, profile, conference):
+        no_exclusion = True
+        if held_tickets:
+            for exclusion in TicketingExclusion.objects.filter(condition=self):
+                no_exclusion = (no_exclusion and exclusion.is_excluded(
+                    held_tickets))
+        if profile:
+            for exclusion in RoleExclusion.objects.filter(condition=self):
+                no_exclusion = (no_exclusion and exclusion.is_excluded(
+                    profile,
+                    conference))
+        return no_exclusion
+
+
+class TicketingEligibilityCondition(EligibilityCondition):
+    '''
+    This is the implementation of the condition under which we give a
+    checklist item to a purchaser because they have purchased a ticket
+    Tickets are realized as BPT Events, the various tickets within an
+    event do not qualify a user for anything more.
+
+    Ticket conditions are additive.  X purchases = X items given to the
+    buyer
+    '''
+    tickets = models.ManyToManyField(TicketItem,
+                                     blank=False)
+
+    def __unicode__(self):
+        unicode_string = ""
+        first = True
+        for ticket in self.tickets.all():
```

Comment:
I'm not following the syntax of that or where in the code to put it?  

    ", ".join(self.tickets.all())

Seems like a nice way to avoid the for loop, but I get failures when I try to paste that in as the return value.  I'm sure I'm missing something... but when I try to dig around for documentation on the join function, I don't find a whole lot out there.